### DISCUSSION TITLE
How to delete nmi pods

### DISCUSSION THREAD

**Discussion by nitin0881** [Azure/aad-pod-identity#1324] [Q&A] [answered]
I tried to delete nmi pods by using 
kubectl delete pod  nmi-pod-name 

and deleting the demon process by using
kubectl delete ds nmi

but the nmi pods comeback again after sometime. Is there any other step to be done. I want to delete and re-setup the nmi pods

**Comment by aramase** [comment, marked as answer]
> but the nmi pods comeback again after sometime. Is there any other step to be done. I want to delete and re-setup the nmi pods

As answered in the other thread, you have the [AKS Pod Managed identity add-on](https://learn.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity) enabled. If you don't need to use pod identity, then disable the add-on and the pods will be deleted. While the add-on is enabled, you can't delete the daemonset.

**Reply by nitin0881** [reply]
Yes the nmi pods were running in managed mode. I had to run the command which deleted the nmi pods and also removed all AKS identities
az aks update -g <<Resource Group Name>> -n <<AKS Service Name>> --disable-pod-identity

Is this the correct way to disable the add-on?

**Reply by aramase** [reply]
Yes!

**Reply by nitin0881** [reply]
Thanks
